client.go

```diff
@@ -184,6 +192,22 @@ func (c Client) GetPrimaryUUID() (uuid string, err error) {
 	return "", err
 }
 
+// DeleteDataset performs a delete request to the given datasetID
+func (c *Client) DeleteDataset(datasetID string) error {
```

Comment:
:question: Given our conversations in Slack I guess we'll aim to use pointers in all our method receivers....although personally I'd rather the client was immutable to avoid any confusing / buggy state mutation.